### DISCUSSION TITLE
So much Share rejected

### DISCUSSION THREAD

**Discussion by James1847** [coinfoundry/miningcore#1143] [Q&A]
![image](https://user-images.githubusercontent.com/29943599/151671104-09b6aab2-f413-4f36-b0ec-5ee034c609ac.png)
It looks runs well at first 3 minutes, but after 3 minutes, more and more share rejected information shows in the log, what happend?
My config as below:
![image](https://user-images.githubusercontent.com/29943599/151671184-38900fe7-f520-46f9-a155-33ac1477b33d.png)

**Comment by XIII39** [comment]
I am finding the error with NiceHash as well.
I created another port with a difficulty of 16 and disabled varDiff and the "low difficulty shares" stopped.

```
"ports": {
    "6060": {
          "listenAddress": "*",
          "name": "GPU",
          "difficulty": 4,
          "varDiff": {
             "minDiff": 4,
             "maxDiff": null,
             "targetTime": 15,
             "retargetTime": 90,
             "variancePercent": 30 
          },
    "6161": {
          "listenAddress": "*",
          "name": "NH",
          "difficulty": 16,
          "varDiff": null
          },
    }
```

It seems that whenever the varDiff recalculates it starts throwing the error even though the shares are greater than the new difficulty set by varDiff..

**Reply by James1847** [reply]
Thank you so much, I will try it right now

**Reply by oliverw** [reply]
> I am finding the error with NiceHash as well. I created another port with a difficulty of 16 and disabled varDiff and the "low difficulty shares" stopped.

You don't need another port to mine with fixed diff. Just configure your miner to use password  `d=<diff>`

**Reply by James1847** [reply]
> > I am finding the error with NiceHash as well. I created another port with a difficulty of 16 and disabled varDiff and the "low difficulty shares" stopped.
> 
> You don't need another port to mine with fixed diff. Just configure your miner to use password `d=<diff>`

His method worked for me, but I will try your method next time, thank you so much

**Reply by XIII39** [reply]
> You don't need another port to mine with fixed diff. Just configure your miner to use password `d=<diff>`

So if we set the password to D=X that would ignore any varDiff settings in the config.json and use the static difficulty of X?
Good to know.

**Reply by oliverw** [reply]
> > You don't need another port to mine with fixed diff. Just configure your miner to use password `d=<diff>`
> 
> So if we set the password to D=X that would ignore any varDiff settings in the config.json and use the static difficulty of X? Good to know.

Yes. 

https://github.com/oliverw/miningcore/blob/master/src/Miningcore/Blockchain/Bitcoin/BitcoinPool.cs#L128

works the same for all other supported blockchains.

**Reply by PromediaB** [reply]
We are using difficulty 4 in D=4 :

and here is our configuration : 

      "enableDaemonWebsocketStreaming": false,
      "blockRefreshInterval": 50,
      "clientConnectionTimeout": 600,
      "banning": {
        "enabled": true,
        "time": 600,
        "invalidPercent": 50,
        "checkThreshold": 50
      },
      "ports": {
        "6161": {
          "listenAddress": "0.0.0.0",
          "difficulty": 4,
            "varDiff": {
              "minDiff": 0.1,
              "maxDiff": null,
              "targetTime": 15,
              "retargetTime": 90,
              "variancePercent": 30


But we have a very low block reward or maybe nothinng for 15 Th Hash.

please could you help us.

**Reply by XIII39** [reply]
15TH with the current network hashrate can take anywhere from 7 - 81 minutes to find a solution, mathematically.
Usually when you use a hashrate calculators online and they says things like "15TH = 15 minutes to a block" the reality is that at the 15 minute point there is a;
37% chance of 0 blocks
37% chance of 1 block
18% chance of 2 blocks
6% chance of 3 blocks
1% chance of 4 blocks

The time to 0% chance for 0 blocks is 81 minutes.
Some days are better than others but there is noting we can do but wait for a GPU to find a solution.
Over a long term, say one year, you will find you are averaging the block per 15 minutes (as long as the network hashrate doesn't double again) but, some days are better than others.

Having a ton of hashrate is great but remember, the GPUs are just guessing. The more GPUs, the more guesses per second you have but there is no guarantee.

A block solution is just a GPU making an acceptable guess on the right piece of work and getting the block populated/accepted though the nodes as fast as possible.

You can have 1 GPU or 100GPUs, at the end of each round only 1 GPU will find the solutions.. A single Hash among the millions on the network.. Well occasionally 2 or more will find a solution but then we're into the reason for Uncle blocks and that's only about 3% of the time.
